crates/bevy_ecs/src/storage/sparse_set.rs

```diff
@@ -3,11 +3,13 @@ use crate::{
     entity::Entity,
     storage::BlobVec,
 };
-use std::{cell::UnsafeCell, marker::PhantomData};
+use std::{cell::UnsafeCell, marker::PhantomData, mem::MaybeUninit};
+
+const PAGE_SIZE: usize = 64;
 
 #[derive(Debug)]
 pub struct SparseArray<I, V = I> {
-    values: Vec<Option<V>>,
+    values: Vec<Option<Box<[Option<V>; PAGE_SIZE]>>>,
```

Comment:
This needs a comment linking back to the PR; the details are relatively arcane and stumbling across this type will be intimidating.